From 750d338864a92521f4daec993af7ef0aba9d609b Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Sun, 21 Apr 2013 21:54:11 +0000
Subject: [PATCH] IO-356 Fix failures in testIO_356_Loop_UTF16 and
 testIO_356_Loop_UTF8 Reset the byte buffer and the encoder together

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/io/trunk@1470353 13f79535-47bb-0310-9956-ffa450edef68
---
 .../org/apache/commons/io/input/CharSequenceInputStream.java    | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/main/java/org/apache/commons/io/input/CharSequenceInputStream.java b/src/main/java/org/apache/commons/io/input/CharSequenceInputStream.java
index ea7e68646be..84ac70af233 100644
--- a/src/main/java/org/apache/commons/io/input/CharSequenceInputStream.java
+++ b/src/main/java/org/apache/commons/io/input/CharSequenceInputStream.java
@@ -210,6 +210,8 @@ public synchronized void reset() throws IOException {
         if (this.mark != NO_MARK) {
             this.cbuf.position(this.mark);
             this.mark = NO_MARK;
+            this.bbuf.limit(0);
+            this.encoder.reset();
         }
     }
 
